Zoo.cs

```diff
@@ -0,0 +1,313 @@
+                
+                // add finished times
+                if (!workerHours.ContainsKey(time + workingTime))
+                    workerHours.Add(time + workingTime, new List<Tuple<Worker, Area, string>>());
+                workerHours[time + workingTime].Add(new Tuple<Worker, Area, string>(worker, area, "finished"));
+            }
+
+            // sort that "finished" statuses come before "started" statuses
+            foreach (int timeidx in workerHours.Keys)  
+            {
+                workerHours[timeidx] = workerHours[timeidx].OrderBy(x => x.Item3).ToList();
+            }
+        }   
+        return workerHours;
+    }
+
+
+    public delegate void WorkStatus(Worker worker, Area area, string status);
+
+    private static void PrintStatusWork(Worker worker, Area area, string status)
+    {
+        Manager.AddReport($"Worker {worker.GetName()} {status} working in {area}");
+    }
+
+    private void PauseToursInArea(Area area)
+    {
+        foreach(Tour tour in this.OngoingTours)
+        {
+            if (tour.Area == area && tour.IsOngoing)
+                tour.StopTour();
+        }
+    }
+
+    private bool IsStoppingWorkerInArea(Area area)
+    {
+        List<Worker> stoppingWorkers = new List<Worker>();
+        
+        stoppingWorkers.AddRange(this.Doctors);
+        stoppingWorkers.AddRange(this.Cleaners);
+        foreach(Worker stoppingWorker in stoppingWorkers)
+        {
+            if (stoppingWorker.Area == area)
+                return true;
+        }
+        return false;
+    }
+
+    private void ResumeTours()
+    {
+        Dictionary<Area, bool> freeAreas = new Dictionary<Area, bool>();
+        foreach(Area area in areas)
+            freeAreas[area] = !IsStoppingWorkerInArea(area);
+
+        foreach(Tour tour in this.OngoingTours)
+        {
+            if ((!tour.IsOngoing && freeAreas[tour.Area]) || tour.IsFinished)
+                tour.ResumeTour();
+        }
+    }
+
+    private int AreaTicketFee(Area area)
+    {
+        const int NormalTicketFee = 50;
+        List<Worker> workers = new List<Worker>();
+        
+        workers.AddRange(this.Doctors);
+        workers.AddRange(this.Cleaners);
+        workers.AddRange(this.Feeders);
+        
+        int WorkerPresentTicketFee = 0;
+        foreach(Worker worker in workers)
+        {
+            if (worker.Area == area)
+                WorkerPresentTicketFee = Math.Max(WorkerPresentTicketFee, worker.WorkerTicketFee);
+        }
+        return WorkerPresentTicketFee == 0? NormalTicketFee : WorkerPresentTicketFee;
+    }
+    public void CreateTour(Area area)
+    {
+        Random rand = new Random();
+        int NUM_OF_VISITORS = rand.Next(4,7);
+
+        List<Visitor> visitors = new List<Visitor>();
+        for(int i=0; i<NUM_OF_VISITORS; i++)
+            visitors.Add(new Visitor($"Visitor{i+1}"));
+        Tour tour = new Tour(new List<Visitor>(), area, Manager);
```

Comment:
Empty line before and after any command with {}. For example- for, if , while , lock, etc.